cmd/launcher/doctor.go

```diff
@@ -0,0 +1,625 @@
+func checkupConnectivity(logger log.Logger, k types.Knapsack) (string, error) {
+	var failures int
+	checkpointer := checkpoint.New(logger, k)
+	connections := checkpointer.Connections()
+	for k, v := range connections {
+		if v == "successful tcp connection" {
+			pass(fmt.Sprintf("%s - %s", k, v))
+		} else {
+			fail(fmt.Sprintf("%s - %s", k, v))
+			failures = failures + 1
+		}
+	}
+
+	ipLookups := checkpointer.IpLookups()
+	for k, v := range ipLookups {
+		valStrSlice, ok := v.([]string)
+		if ok && len(valStrSlice) > 0 {
+			pass(fmt.Sprintf("%s - %s", k, valStrSlice))
+		} else {
+			fail(fmt.Sprintf("%s - %s", k, valStrSlice))
+			failures = failures + 1
+		}
+	}
+
+	notaryVersions, err := checkpointer.NotaryVersions()
+	if err != nil {
+		fail(fmt.Errorf("could not fetch notary versions: %w", err))
+		failures = failures + 1
+	}
+
+	for k, v := range notaryVersions {
+		// Check for failure if the notary version isn't a parsable integer
+		if _, err := strconv.ParseInt(v, 10, 32); err == nil {
+			pass(fmt.Sprintf("%s - %s", k, v))
+		} else {
+			fail(fmt.Sprintf("%s - %s", k, v))
+			failures = failures + 1
+		}
+	}
+
+	if failures == 0 {
+		return "Successfully communicated with Kolide", nil
+	}
+
+	return "", fmt.Errorf("%d failures encountered while attempting communication with Kolide", failures)
+}
+
+// checkupVersion tests to see if the current launcher version is up to date
+func checkupVersion(v version.Info) (string, error) {
+	info(fmt.Sprintf("Current version: %s", v.Version))
+	// TODO: Query TUF for latest available version for this launcher instance
+	warn(fmt.Sprintf("Target version: %s", "Unknown"))
+
+	// TODO: Choose failure based on current >= target
+	if true {
+		return "Up to date!", nil
+	}
+
+	return "", fmt.Errorf("launcher is out of date")
+}
+
+// checkupConfigFile tests that the config file is valid and logs it's contents
+func checkupConfigFile(filepath string) (string, error) {
+	file, err := os.Open(filepath)
+	if err != nil {
+		return "", fmt.Errorf("No config file found")
+	}
+	defer file.Close()
+
+	// Parse the config file how launcher would
+	err = ff.PlainParser(file, func(name, value string) error {
+		info(fmt.Sprintf("%s %s", name, value))
+		return nil
+	})
+
+	if err == nil {
+		return "Config file found", nil
+	}
+	return "", fmt.Errorf("Invalid config file")
+}
+
+// checkupLogFiles checks to see if expected log files are present
+func checkupLogFiles(filepaths []string) (string, error) {
+	var foundCurrentLogFile bool
+	for _, f := range filepaths {
+		filename := filepath.Base(f)
+		info(filename)
+
+		if filename == "debug.json" {
+			foundCurrentLogFile = true
+
+			fi, err := os.Stat(f)
+			if err == nil {
+				info("")
+				info(fmt.Sprintf("Most recent log file: %s", filename))
+				info(fmt.Sprintf("Latest modification: %s", fi.ModTime().String()))
+				info(fmt.Sprintf("File size (B): %d", fi.Size()))
+			}
+		}
+	}
```

Comment:
lets try to reduce some nesting:
```suggestion
	for _, f := range filepaths {
		filename := filepath.Base(f)
		info(filename)
		
		if filename != "debug.json" {
		    continue
		}

	        foundCurrentLogFile = true
        
	        fi, err := os.Stat(f)
	        if err != nil {
	            continue
	        }

		info("")
		info(fmt.Sprintf("Most recent log file: %s", filename))
		info(fmt.Sprintf("Latest modification: %s", fi.ModTime().String()))
		info(fmt.Sprintf("File size (B): %d", fi.Size()))
	}
```